tidy3d/web/cli/converter.py

```diff
@@ -0,0 +1,1340 @@
+        elif Lines[i][5:10] == "y_max":
+            y_max = Lines[i][12:].split(";")[0][:-1]
+            yl = y_max + "-" + y_min
+        elif Lines[i][5:10] == "z_min":
+            z_min = Lines[i][12:].split(";")[0][:-1]
+            zl = z_max + "-" + z_min
+        elif Lines[i][5:10] == "z_max":
+            z_max = Lines[i][12:].split(";")[0][:-1]
+            zl = z_max + "-" + z_min
+        elif Lines[i][5:11] == "x span":
+            xl = Lines[i][13:].split(";")[0][:-1]
+        elif Lines[i][5:11] == "y span":
+            yl = Lines[i][13:].split(";")[0][:-1]
+        elif Lines[i][5:11] == "z span":
+            zl = Lines[i][13:].split(";")[0][:-1]
+        elif Lines[i][5:19] == "mode selection":
+            mode = Lines[i][22:].split(";")[0][:-1]
+            if mode == "'fundamental TM mode'":
+                mode = "1"
+                num_modes = "2"
+            else:
+                mode = "0"
+        elif Lines[i][5:22] == "wavelength center":
+            wave_center = Lines[i][24:].split(";")[0][:-1] + "*1e6"
+        elif Lines[i][5:20] == "wavelength span":
+            wave_span = Lines[i][22:].split(";")[0][:-1] + "*1e6"
+        elif Lines[i][5:21] == "frequency points":
+            num_freqs = Lines[i][23:].split(";")[0][:-1]
+            freqs = (
+                "np.linspace("
+                + wave_center
+                + "-"
+                + wave_span
+                + "/2,"
+                + wave_center
+                + "+"
+                + wave_span
+                + "/2,"
+                + num_freqs
+                + ")"
+            )
+        i += 1
+        if i == len(Lines):
+            break
+    mode_monitor_string += num_modes + ") # check if solving for correct mode!\n"
+    mode_monitor_string += name.replace(" ", "_") + " = td.ModeMonitor(\n"
+    x, y, z, xl, yl, zl = _to_um([x, y, z, xl, yl, zl])
+    mode_monitor_string += "\tcenter=(" + x + "," + y + "," + z + "),\n"
+    mode_monitor_string += "\tsize=(" + xl + "," + yl + "," + zl + "),\n"
+    mode_monitor_string += "\tfreqs=" + freqs + "\n"
+    mode_monitor_string += "\tmode_spec=" + mode_spec_name + ",\n"
+    mode_monitor_string += "\tname='" + name + "',\n"
+    mode_monitor_string += ")\n"
+    return mode_monitor_string, name.replace(" ", "_")
+
+
+def lsf_reader(filename=str) -> None:
+    tidy3d_file = "import numpy as np\n"
+    tidy3d_file += "import matplotlib.pyplot as plt\n"
+    tidy3d_file += "import tidy3d as td\n"
+    tidy3d_file += "import tidy3d.web as web\n"
+    tidy3d_file += "\n"
+
+    structures = "["
+    rect, sph, cyl, polyslab = 0, 0, 0, 0
+    sources, modesrc, planewv, ptdipole, gauss, tfsf = "[", 0, 0, 0, 0, 0
+    monitors = "["
+    modemon, fluxmon, fieldmon = 0, 0, 0
+    gridspec_string = "grid_spec=td.GridSpec.auto()"
+    override_structures, overstrct = "[", 0
+
+    if filename[-4:] != ".lsf":
+        print("Needs .lsf format")
+    try:
```

Comment:
this can be handled simply as 

```py
with open(filename, "r") as file_lsf:
    Lines = file_lsf.readlines()
    ... # rest of code
```

it will take care of the file errors and also closing the file at the end or if anything goes wrong.

https://realpython.com/working-with-files-in-python/